Sources/SwiftDocC/Model/Identifier.swift

```diff
@@ -141,10 +141,14 @@ extension TopicReferenceResolutionErrorInfo {
 /// > its data.
 public struct ResolvedTopicReference: Hashable, Codable, Equatable, CustomStringConvertible {
     typealias ReferenceBundleIdentifier = String
-    typealias ReferenceKey = String
+    private struct ReferenceKey: Hashable {
+        var path: String
+        var fragment: String?
+        var sourceLanguages: Set<SourceLanguage>
+    }
     
     /// A synchronized reference cache to store resolved references.
-    static var sharedPool = Synchronized([ReferenceBundleIdentifier: [ReferenceKey: ResolvedTopicReference]]())
+    private static var sharedPool = Synchronized([ReferenceBundleIdentifier: [ReferenceKey: ResolvedTopicReference]]())
```

Comment:
Making this private leads to adding `_numberOfCachedReferences` and making all of the test code changes below? Seems like a good idea to encapsulate the synchronization and other details.

Reply:
Yes. I feel like the shared pool itself should be a hidden implementation detail. 

The tests only needed to access the pool to check the number of cached elements so I added a dedicated function for that instead so that the pool in general didn't need to be exposed to the rest of the code.